tests/test_dist_save_load.py

```diff
@@ -0,0 +1,169 @@
+"""Test distributed mulit-host device mesh."""
+
+import os
+import unittest
+import subprocess
+import re
+import tensorstore as ts
+
+from flax import linen as nn
+from flax import optim
+import jax
+import jax.numpy as jnp
+from jax.interpreters import pxla
+import numpy as np
+import ray
+
+from alpa import DeviceCluster, parallelize, set_parallelize_options, testing, DistributedArray
+from alpa.testing import assert_allclose
+
+
+class DeviceMeshTest(unittest.TestCase):
+
+    def setUp(self):
+        os.environ["XLA_PYTHON_CLIENT_ALLOCATOR"] = "platform"
+        ray.init(address="auto", ignore_reinit_error=True)
+
+    def tearDown(self):
+        ray.shutdown()
+    
+    def test_distributed_array_save_load(self):
```

Comment:
This test only works for an 8-GPU cluster. Change this to work for more general clusters or add an if to skip this test for clusters of other shapes.